openedx/core/djangoapps/util/management/commands/run_aggregator.py

```diff
@@ -0,0 +1,272 @@
+                self._create_block(parent=vertical, category='html')
+                self.store.publish(vertical.location, ModuleStoreEnum.UserID.test)
+
+        reset_queries()
+        time_taken = self._time_handler(course_published_handler, course_key=self.course.id)
+        self._copy_executed_queries()
+
+        self._print_results_header(u"test_course_published_handler_when_block_is_added", time_taken=time_taken)
+        self._assert_vertical_completion_for_all_users(
+            vertical, self.course_breadth[3] / (self.course_breadth[3] + 1.0)
+        )
+        self._print_results_footer()
+
+    def test_item_deleted_handler_when_block_is_deleted(self):
+
+        # Other listeners are connected so we time the handler alone later.
+        from completion_aggregator.signals import item_deleted_handler
+        SignalHandler.course_published.disconnect(item_deleted_handler)
+
+        vertical = self.course.get_children()[-1].get_children()[-1].get_children()[-1]
+        self._complete_blocks_for_users(vertical.get_children()[1:], self.users)
+        self._assert_vertical_completion_for_all_users(
+            vertical, (self.course_breadth[3] - 1.0) / self.course_breadth[3]
+        )
+        block = vertical.get_children()[0]
+        with self.store.branch_setting(ModuleStoreEnum.Branch.draft_preferred, self.course.id):
+            self.store.delete_item(block.location, ModuleStoreEnum.UserID.test)
+
+        reset_queries()
+        time_taken = self._time_handler(item_deleted_handler, usage_key=block.location, user_id=None)
+        self._copy_executed_queries()
+
+        self._print_results_header(u"test_item_deleted_handler_when_block_is_deleted", time_taken=time_taken)
+        self._assert_vertical_completion_for_all_users(vertical, 1.0)
+        self._print_results_footer()
+
+    def test_individual_block_completions(self):
+
+        from completion_aggregator.models import Aggregator
+
+        times_taken = []
+
+        for user in self.users:
+            user.next_block_index_to_complete = 0
+
+        users = list(self.users)
+
+        reset_queries()
+
+        for __ in range(self.completions_count):
+            random_user = random.choice(users)
+            next_block = self.blocks[random_user.next_block_index_to_complete]
+            random_user.next_block_index_to_complete += 1
+            if random_user.next_block_index_to_complete >= len(self.blocks):
+                users.remove(random_user)
+
+            timer_start = self.timer()
+            self._complete_blocks_for_users([next_block], [random_user])
+            timer_end = self.timer()
+            elapsed_milliseconds = (timer_end - timer_start)
+            times_taken.append(elapsed_milliseconds)
+
+        self._copy_executed_queries()
+
+        for user in self.users:
+            expected_verticals_completed = math.floor(user.next_block_index_to_complete / self.course_breadth[3])
```

Comment:
Consider using `//`instead of `math.floor`.

Reply:
Done.